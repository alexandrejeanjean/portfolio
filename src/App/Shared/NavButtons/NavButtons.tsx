import React from 'react'
import classNames from 'classnames'
import { Link } from 'react-router-dom'
import '../../Screens/Navigation/navigation.scss'

type Props = { text: string; position: string; path: string }

const NavButtons = ({ text, position, path }: Props) => {
  let btnClasses = classNames({
    'nav-button-link': true,
    left: position === 'left',
    right: position === 'right',
    bottom: position === 'bottom',
    focused:
      (path === '/projets' && position === 'left') ||
      (path === '/services' && position === 'right'),
    'bottom-focused': path === '/a-propos' && position === 'bottom',
  })

  let decorationClass = classNames({
    'nav-decoration-line': true,
    'deco-left': position === 'left',
    'deco-right': position === 'right',
    'deco-bottom': position === 'bottom',
  })

  const getDestination = () => {
    if (text === 'Faisons connaissance') return '/a-propos'
    if (text === 'Projets') return '/projets'
    if (text === 'Mes services') return '/services'
    return '/'
  }

  return (
    <div className='nav-link-wrapper'>
      <Link className={btnClasses} to={getDestination()}>
        {text}
      </Link>
      <span className={decorationClass}></span>
    </div>
  )
}

export default NavButtons
