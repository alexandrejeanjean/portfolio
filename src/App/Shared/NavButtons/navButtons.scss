@import "../../constants/colors.scss";

.desktop-nav-link-wrapper {
  .nav-button-link {
    position: absolute;
    font-size: 0.8em;
    text-transform: uppercase;
    text-align: center;
    letter-spacing: 0.2em;
    margin: 0;
    text-decoration: none;
    color: $main;
    transition: all 0.2s ease;

    &:hover {
      font-weight: bold;
    }
  }

  .focused,
  .bottom-focused {
    font-weight: bold;
  }

  .focused.nav-button-link::after {
    content: "";
    width: 0%;
    height: 100%;
    background: url("../../Assets/underline.svg") no-repeat;
    position: absolute;
    top: 0.2em;
    left: -0.5em;
    z-index: -1;
    display: inline-block;
    animation-name: underlineAnimate;
    animation-duration: 0.2s;
    animation-fill-mode: both;
  }

  .left {
    left: 0vw;
    top: 50vh;
    transform: rotate(-90deg);
  }

  .right {
    right: -2vw;
    top: 50vh;
    transform: rotate(-90deg);
  }

  .bottom {
    left: 0;
    right: 0;
    bottom: 5vh;
  }

  .bottom-focused.nav-button-link::after {
    content: "";
    width: 140%;
    height: 0%;
    background: url("../../Assets/underline.svg") no-repeat;
    position: absolute;
    left: 46%;
    top: 10%;
    z-index: -1;
    animation-name: BottomUnderlineAnimate;
    animation-duration: 0.2s;
    animation-fill-mode: both;
  }

  .nav-decoration-line {
    position: absolute;
    display: block;
    width: 0.1em;
    height: 1.2em;
    background: linear-gradient(259.07deg, #daf8ff 4.07%, #b2eaf8 100%);
  }

  .deco-left {
    left: 0.4vw;
    top: 50vh;
    transform: rotate(-90deg);
  }

  .deco-right {
    right: 0.4vw;
    top: 50vh;
    transform: rotate(-90deg);
  }

  .deco-bottom {
    left: 50vw;
    bottom: 0vh;
  }
}

@media screen and (max-width: 768px) {
  .desktop-nav-link-wrapper {
    display: none;
  }

  .mobile-nav-link-wrapper {
    width: 30%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: space-evenly;

    .mobile-nav-link {
      height: 90%;
      width: 100%;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;

      .mobile-nav-icon {
        width: 2em;
        margin-bottom: -0.5em;
        margin-top: 1em;
      }

      .mobile-nav-link-text {
        font-size: 0.7em;
        text-align: center;
      }
    }
  }
  .mobile-focused {
    font-weight: bold;
    background-color: rgba(158, 158, 158, 0.1);
    border-radius: 2em;
  }
}
