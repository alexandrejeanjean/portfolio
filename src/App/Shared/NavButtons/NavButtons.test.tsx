import React from 'react'
import { shallow } from 'enzyme'
import NavButtons from './NavButtons'

describe('NavButtons', () => {
  it('should render nav buttons correctly', () => {
    const navButton = shallow(
      <NavButtons
        text='Faisons connaissance'
        position='bottom'
        path='/a-propos'
      />
    )
    expect(navButton).toMatchSnapshot()
  })
})
