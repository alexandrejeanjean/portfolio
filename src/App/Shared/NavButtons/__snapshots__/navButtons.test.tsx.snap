// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButtons should render nav buttons correctly 1`] = `
<div
  className="nav-link-wrapper"
>
  <Link
    className="nav-button-link bottom bottom-focused"
    to="/a-propos"
  >
    Faisons connaissance
  </Link>
  <span
    className="nav-decoration-line deco-bottom"
  />
</div>
`;
